docs/basics/primitives.md

```diff
@@ -5,30 +5,195 @@ to describe the set of types that are native to its platform, we leverage and ov
+Use them when you really want something to be optional.
+The outcome is that instead of a ```T``` you get an ```Option[T]``` and you can ```match, fold, flatMap, map``` on a ```None```.
+
+The ```Optional``` part is handled at a DSL level, it's not translated to Cassandra in any way.
+
+| phantom columns               | Java/Scala type                   | Cassandra columns |
+| ---------------               | -------------------------         | ----------------- |
+| OptionalBlobColumn            | Option[java.nio.ByteBuffer]       | blog              |
+| OptionalBigDecimalColumn      | Option[scala.math.BigDecimal]     | decimal           |
+| OptionalBigIntColumn          | Option[scala.math.BigInt]         | varint            |
+| OptionalBooleanColumn         | Option[scala.Boolean]             | boolean           |
+| OptionalDateColumn            | Option[java.util.Date]            | timestamp         |
+| OptionalDateTimeColumn        | Option[org.joda.time.DateTime]    | timestamp         |
+| OptionalDoubleColumn          | Option[scala.Double]              | double            |
+| OptionalEnumColumn            | Option[scala.Enumeration]         | text              |
+| OptionalFloatColumn           | Option[scala.Float]               | float             |
+| OptionalIntColumn             | Option[scala.Int]                 | int               |
+| OptionalInetAddressColumn     | Option[java.net.InetAddress]      | inet              |
+| OptionalLongColumn            | Option[Long]                      | long              |
+| OptionalStringColumn          | Option[java.lang.String]          | text              |
+| OptionalUUIDColumn            | Option[java.util.UUID]            | uuid              |
+| OptionalTimeUUID              | Option[java.util.UUID]            | timeuuid          |
+
+It is also possible to express your entire table logic using just `Col` if you
+prefer it. Because it is `Primitives` that do the real heavy lifting, the below
+are equivalent:
+
+```scala
+
+class Recipes extends Table[Recipes, Recipe] {
+
+  object url extends StringColumn with PartitionKey
+
+  object description extends OptionalStringColumn
+
+  object ingredients extends ListColumn[String]
+
+  object servings extends OptionalIntColumn
+
+  object lastcheckedat extends DateTimeColumn
+
+  object props extends MapColumn[String, String]
+
+  object uid extends UUIDColumn
+}
+```
+
+The same can be achieved by writing the below code:
+
+```scala
+class Recipes extends Table[Recipes, Recipe] {
+
+  object url extends Col[String] with PartitionKey
+
+  object description extends OptionalCol[String]
+
+  object ingredients extends Col[List[String]]
+
+  object servings extends OptionalCol[Int]
+
+  object lastcheckedat extends Col[DateTime]
+
+  object props extends Col[Map[String, String]]
+
+  object uid extends Col[UUID]
+}
+```
+
+There is virtually no difference whatsoever in doing things one way or the other,
```

Comment:
remove this paragraph for:

How:
Both of the above examples use the same macro which generate all Cassandra I/O required to marshall types across the wire to and from Cassandra.